nncf/experimental/common/pruning/operations.py

```diff
@@ -0,0 +1,638 @@
+                offset=first_block.offset,
+                pruning_dimension=1,
+                closed_branches=1
+            )
+            group.add_block(consumer_block)
+
+
+class BatchNormPruningOp(BasePruningOp):
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        identity_mask_propagation(node, graph)
+
+
+class GroupNormPruningOp(BasePruningOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode) -> bool:
+        # For Instance Normalization
+        return isinstance(node.layer_attributes, GroupNormLayerAttributes) \
+            and node.layer_attributes.num_groups == node.layer_attributes.num_channels
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        if cls.accept_pruned_input(node):
+            identity_mask_propagation(node, graph)
+        else:
+            node.data['output_mask'] = None
+
+
+class ConcatPruningOp(BasePruningOp):
+    @classmethod
+    def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph,
+                             tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> Optional[NNCFTensor]:
+        """
+        Generate output mask from input masks with all None replaced by identity masks.
+        If all input masks is None return None.
+
+        :param node: Node to determine it's sources.
+        :param graph: NNCF graph to work with.
+        :param tensor_processor: Interface with tensor processing methods.
+        :return: Filled input masks.
+        """
+        input_edges = graph.get_input_edges(node)
+        previous_nodes = [edge.from_node for edge in input_edges]
+        input_masks = [input_node.data['output_mask'] for input_node in previous_nodes] #  type: List[NNCFTensor]
+        not_empty_masks = [mask for mask in input_masks if mask is not None]
+        if not not_empty_masks:
+            return None
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        result_mask = cls.generate_output_mask(node, graph, tensor_processor)
+        node.data['output_mask'] = result_mask
+
+
+class ElementwisePruningOp(BasePruningOp):
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        input_masks = get_input_masks(node, graph)
+        if not input_masks:
+            input_masks = [None]
+
+        # elif all(m is None for m in input_masks):
+        output_mask = input_masks[0]
+        if len(input_masks) == 1:
+            nncf_logger.warning(f"ElementWise with a single input is not properly supported. node_name={node.node_name}"
```

Comment:
```suggestion
            nncf_logger.warning(f"ElementWise with a single input is not properly supported. "
```
Added by mistake?

Reply:
No, `node_name` should provide useful context when the problematic operation in the graph.